### PR TITLE
Added new TC DeleteBackupOfUserSharedRBAC

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Added New TC
DeleteBackupOfUserSharedRBAC - https://portworx.testrail.net/index.php?/cases/view/87560

**Which issue(s) this PR fixes** (optional)
Closes #PA-1397

**Special notes for your reviewer**:
This PR doesn't contains 2.5.2 api
logs:

s3:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/2406/

https://aetos.pwx.purestorage.com/resultSet/testSetID/323498

NFS:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/NFS/job/PX-Cloud-Vanilla-NFS/128/

https://aetos.pwx.purestorage.com/resultSet/testSetID/327353

### CODE REVIEW

**Review by snigdha-px — APPROVED** [review]
Can we have log.InfoD for each step
It will help during debugging to know the step which has failed

**Comment by ak-px** [comment]
@snigdha-px , Thanks for reviewing, Addressed all your comments.

File: drivers/backup/portworx/portworx.go [hunk 1]
```diff
@@ -1535,6 +1537,68 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
...
+	}
+	Rules := resp.GetRules()
+	for _, rule := range Rules {
+		ruleNames = append(ruleNames, rule.Name)
+	}
+	return ruleNames, nil
+}
+
+// GetAllSchedulePolicys gets the all SchedulePolicys names for the given org
+func (p *portworx) GetAllSchedulePolicys(ctx context.Context, orgID string) ([]string, error) {
```
Comment (kphalgun-px) [inline drivers/backup/portworx/portworx.go:1585]: nit GetAllSchedulePolicies

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -4171,3 +4172,159 @@ func CreateRuleForBackupWithMultipleApplications(orgID string, appList []string,
 	}
 	return preRuleName, postRuleName, nil
 }
+
+// CreateBackupScheduleIntervalPolicy create periodic schedule policy with give interval.
+func CreateBackupScheduleIntervalPolicy(retian int64, intervalMins int64, incrCount uint64, periodicSchedulePolicyName string, periodicSchedulePolicyUid string, OrgID string, ctx context.Context) (err error) {
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:4177]: Is it WithContext?

File: drivers/backup/portworx/portworx.go [hunk 3]
```diff
@@ -1535,6 +1537,68 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
...
+	var schedulePolicyNames []string
+	schedulePolicyRequest := &api.SchedulePolicyEnumerateRequest{
+		OrgId:  orgID,
+		Labels: map[string]string{},
+	}
+	resp, err := p.EnumerateSchedulePolicy(ctx, schedulePolicyRequest)
+	if err != nil {
+		return schedulePolicyNames, err
```
Comment (kphalgun-px) [inline drivers/backup/portworx/portworx.go:1593]: How about?

`return nil, err`

File: tests/backup/backup_orphaned_delete_test.go [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+			log.InfoD("Create Users with Different types of roles")
+			roles := [3]backup.PxBackupRole{backup.ApplicationOwner, backup.InfrastructureOwner, backup.ApplicationUser}
+			for i := 1; i <= numOfUsers/3; i++ {
+				for _, role := range roles {
+					userName := fmt.Sprintf("testauser-%v-%s", role, RandomString(4))
+					firstName := fmt.Sprintf("FirstName-%v-%s", role, RandomString(4))
+					lastName := fmt.Sprintf("LastName-%v-%s", role, RandomString(4))
+					email := fmt.Sprintf("testuser-%v-%s@cnbu.com", role, RandomString(4))
+					err := backup.AddUser(userName, firstName, lastName, email, commonPassword)
+					log.FailOnError(err, "Failed to create user - %s", userName)
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:107]: How about?

userName := createUsers(1)[0] just to be consistent?

File: tests/backup/backup_orphaned_delete_test.go [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+				log.Infof("Pre backup rule [%s] uid: [%s]", preRuleName, preRuleUid)
+			}
+			if postRuleName != "" {
+				postRuleUid, err = Inst().Backup.GetRuleUid(orgID, ctx, postRuleName)
+				log.FailOnError(err, "Fetching post backup rule [%s] uid", postRuleName)
+				log.Infof("Post backup rule [%s] uid: [%s]", postRuleName, postRuleUid)
+			}
+		})
+
+		Step("update ownership for RBAC resource with non-admin users from px-admin", func() {
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:164]: How about? "U"pdate

File: tests/backup/backup_orphaned_delete_test.go [hunk 6]
```diff
@@ -0,0 +1,605 @@
...
+					log.FailOnError(err, "Failed to add role for user - %s", userName)
+					userNames = append(userNames, userName)
+					userUID, err := backup.FetchIDOfUser(userName)
+					log.FailOnError(err, "Failed to fetch uid for - %s", userName)
+					userIdMap[userName] = userUID
+				}
+			}
+		})
+		Step(fmt.Sprintf("Adding Credentials and Registering Backup Location from px-admin user"), func() {
+			log.InfoD(fmt.Sprintf("Creating cloud credentials and backup location from px-admin user"))
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:118]: How about? ~~Registering~~

**Comment by kphalgun-px** [comment]
The test runs do not seem to be correct. In S3, the test 'DeleteBackupOfUserNonSharedRBAC' passed, but it failed in NFS. Additionally, the test names are different.

**Comment by ak-px** [comment]
@kphalgun-px , Update the correct logs and addressed all your comments , Please re-review it.

**Comment by ak-px** [comment]
Thanks for the reviews. These changes are now included in PR https://github.com/portworx/torpedo/pull/1669.

**Pull request closed without merging**